File: June/28-06-2024/The Palindrome Pattern.py
#User function Template for python3

class Solution:
    def pattern (self, arr):
  #Your Code Start
        n = len(arr)
        
        # Helper function to check if a sequence is a palindrome
        def is_palindrome(seq):
            l, r = 0, len(seq) - 1
            while l < r:
                if seq[l] != seq[r]:
                    return False
                l += 1
                r -= 1
            return True

        # Check each row
        for i in range(n):
            if is_palindrome(arr[i]):
                return str(i) + " R"

        # Check each column
        for j in range(n):
            col = [arr[i][j] for i in range(n)]
            if is_palindrome(col):
                return str(j) + " C"
        
        # Return -1 if no palindromic pattern is found
        return "-1"
        #Your code End

#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    ob = Solution()
    t = int(input())
    for _ in range(t):
        n = int(input())
        arr = list(map(int, input().split()))
        a = list()
        c = 0
        for i in range(0, n):
            X = list()
            for j in range(0, n):
                X.append(arr[c])
                c += 1
            a.append(X)
        ans = ob.pattern(a)
        print(ans)

# } Driver Code Ends

File: June/28-06-2024/The Palindrome Pattern.java
//{ Driver Code Starts
// Initial Template for Java

// Initial Template for Java

import java.io.*;
import java.util.*;

class GFG {
    public static void main(String args[]) throws IOException {

        BufferedReader read = new BufferedReader(new InputStreamReader(System.in));

        Solution ob = new Solution();
        int t = Integer.parseInt(read.readLine());
        while (t-- > 0) {
            String input_line[] = read.readLine().trim().split("\\s+");
            int n = Integer.parseInt(input_line[0]);
            String input_line1[] = read.readLine().trim().split("\\s+");
            int[][] a = new int[n][n];
            int c = 0;
            for (int i = 0; i < n; i++) {
                for (int j = 0; j < n; j++) {
                    a[i][j] = Integer.parseInt(input_line1[c]);
                    c++;
                }
            }
            String ans = ob.pattern(a);
            System.out.println(ans);
        }
    }
}

// } Driver Code Ends


// User function Template for Java

class Solution {
    
    //Your Code Start
  
  
     // Helper function to check if a sequence is a palindrome
    private boolean isPalindrome(int[] seq) {
        int l = 0, r = seq.length - 1;
        while (l< r) {
            if (seq[l] != seq[r]) {
                return false;
            }
            l++;
            r--;
        }
        return true;
    }

    public String pattern(int[][] arr) {
        int n = arr.length;

        // Check each row
        for (int i = 0; i < n; i++) {
            if (isPalindrome(arr[i])) {
                return i + " R";
            }
        }

        // Check each column
        for (int j = 0; j < n; j++) {
            int[] col = new int[n];
            for (int i = 0; i < n; i++) {
                col[i] = arr[i][j];
            }
            if (isPalindrome(col)) {
                return j + " C";
            }
        }

        // Return -1 if no palindromic pattern is found
        return "-1";
        
    }
    
}
    //Your Code End
